Automattic/wp-calypso
# Social Auth: Introduce `calypso_social_auth_attempt` event

#### Proposed Changes

The new event `calypso_social_auth_attempt` will differentiate actual errors in the authentication workflow from attempts. It handles both login:

- `unknown_user` when the user tries to social login with an account not yet registered;
- `user_exists` when the user tries to social login with an account that was registered using an e-mail and password;

and sign up:

- `user_exists` when the user tries to social sign up with an account that was registered using an e-mail and password;

#### Testing Instructions

See "Testing Instructions" in https://github.com/Automattic/wp-calypso/pull/64957 to set up a working wpcalypso.wordpress.com environment on your local machine.

Open the DevTools and write `localStorage.setItem('debug', 'calypso:analytics');` so you can see the events being dispatched in the console.

Open https://wpcalypso.wordpress.com/log-in and try to social login with an account not yet registered with WPCOM:
- `calypso_social_auth_attempt` should've been dispatched with
  - `social_account_type: 'google'`
  - `starting_point: 'login'`
  - `error_code: 'unknown_user'`

Open https://wpcalypso.wordpress.com/log-in and try to social login with an account that was registered using an e-mail and password:
- `calypso_social_auth_attempt` should've been dispatched with
  - `social_account_type: 'google'`
  - `starting_point: 'login'`
  - `error_code: 'user_exists'`

Open https://wpcalypso.wordpress.com/start/user and try to social sign-up with an account that was registered using an e-mail and password:
- `calypso_social_auth_attempt` should've been dispatched with
  - `social_account_type: 'google'`
  - `starting_point: 'signup'`
  - `error_code: 'user_exists'`

Related to https://github.com/Automattic/wp-calypso/issues/65896.
